fairscale/nn/misc/flatten_params_wrapper.py

```diff
@@ -364,18 +378,60 @@ def _unflatten_params(self, external_data: Optional[List[Optional[Tensor]]] = No
             delattr(self, n)
         self.flat_params = []
 
+    # The post backward hook used to accumulate fp32 gradients
+    def _grad_accumulation_hook(
+        self,
+        grad,
+        param_index,
+    ):
+        if self.fp32_grads[param_index] is None:
+            self.fp32_grads[param_index] = grad.to(torch.float32)
```

Comment:
It's been a while to look at FSDP code so can be a silly question, but wonder for TransformerEngine modules that keep their weight grads in .main_grad in fp32 precision this can be a duplication.